doc/HttpTransportAdapter.md

```diff
@@ -0,0 +1,192 @@
+# HTTP Transport Adapter
+
+The Azure SDK for C++ creates and sends HTTP requests using an HTTP Transport Adapter. This adapter provides a general abstraction for an HTTP stack (like libcurl, WinHTTP, etc) by defining the interface it needs to call that anyone can implement to customize which HTTP stack gets used.
+
+## Default HTTP Transport Adapter
+The Azure SDK for C++ provides two HTTP transport adapters as part of the azure-core package:
+- The libcurl transport adapter
```

Comment:
Can you please add on platforms these work. For example: libcurl is for Windows, Linux, Mac; WinHTTP is for Windows